files-community/Files
# Code Quality: Introduce details layout columns generator phase 2

## Summary

As of now, you have to add literally 250 lines only for one column when you are adding a new one (you can see [git columns introduction PR](https://github.com/files-community/Files/pull/12485)). We are going to introduce columns generator and this is the step 2.

## Task Checklist

- [x] Removed ColumnsViewModel
- [x] Introduced a collection of DetailsLayoutColumnItem
- [x] Set default layout settings to the app json settings.
- [x] Set new default settings to the app json settings.
- [ ] Generate columns when page content changed.
- [ ] Generate context menu flyout of the details layout header.
- [x] Implemented ContentSizer
- [ ] Implemented DataTable
- [ ] Implemented DataColumn
- [ ] Implemented DataRow

## Steps To Validate Changes

1. Launch the app
2. Go to a folder
3. Resize some columns
4. Hide some columns
5. Go to another folder
6. See the columns you hid are shown in that folder
7. Go to git folder
8. See git columns appear

If nothing unexpected things(crashes and wrong behaviors) happen, you confirmed layout settings for each page are stored correctly in user_settings.db, hide/show operation is working properly, the app is able to fetch settings from the database and use them in an actual layout.

## Known Issues

_N/A_

## PR Checklist

  - #12505
  - #14127

## Screenshots

_N/A_

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do that, instead.
